eyes-it.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+const {eyes, appName} = require('./eyes');
+const merge = require('lodash.merge');
+
+function buildSpecName(spec, version) {
+  var versionDescription = version ? ' version: ' + version : '';
+
+  return 'eyes.it ' + spec.getFullName() + versionDescription;
+}
+
+function handleError(err, done) {
+  fail(err);
+  done();
+}
+
+var _enableSnapshotAtBrowserGet = false;
+
+const originalBrowserGet = browser.get;
+browser.get = function (address) {
+  return originalBrowserGet.apply(this, arguments).then(function (result) {
+    const res = _enableSnapshotAtBrowserGet ? eyes.checkWindow('get ' + address) : Promise.resolve();
+    return res.then(() => result);
+  });
+};
+
+const defaultConfig = {
+    windowSize: eyes.defaultWindowSize,
+    enableSnapshotAtEnd: true,
+    enableSnapshotAtBrowserGet: true,
+}
+
+/**
+ * Call original `it` and modify before and after
+ *
+ * @param {*} _it the original `it` or `fit` function
+ * @param {*} _itArgs the original arguments of the original `it` or `fit` function.
+ * @param {*} config `eyes.it` configuration
+ * @returns spec
+ */
+function eyesIt(_it, _itArgs, config) {
+  const {
+    windowSize,
+    specVersion,
+    enableSnapshotAtBrowserGet,
+    enableSnapshotAtEnd,
+  } = merge({}, defaultConfig, config);
+
+  var spec = _it.apply(this, _itArgs);
+  var hooked = spec.beforeAndAfterFns;
+  spec.beforeAndAfterFns = function () {
+    //TODO: are we sure that we need to pass args (msg, itFn) to hooked?
+    var result = hooked.apply(this, _itArgs);
+    result.befores.unshift({
+      fn: function (done) {
+        _enableSnapshotAtBrowserGet = enableSnapshotAtBrowserGet;
+        eyes.open(browser, appName, buildSpecName(spec, specVersion), windowSize).then(done);
+      }, timeout: () => 30000
+    });
+    result.afters.push(
```

Comment:
This used to be `result.afters.ushift`

Reply:
This is a breaking change. In many cases we reset the state of the page after the test. So This now means that the final screenshot will be different

Reply:
Good point !
We can split the `afters` hooks , and put the `checkWindow` at the beginning,
and the `eyes.close` at the end.
That way everybody wins.